src/pagination/pagination.tsx

```diff
@@ -44,12 +48,15 @@ const Component = (
     const [hoverRightButton, setHoverRightButton] = useState(false);
     const [hoverLeftButton, setHoverLeftButton] = useState(false);
     const [inputText, setInputText] = useState<string>("");
+    const [pageSizeLocal, setPageSize] = useState<number>(pageSize);
+    const [selecctedOption, setSelecctedOption] = useState<DropdownItemProps>();
 
     const boundaryRange = 1;
     const siblingRange = 1;
-    const totalPages = Math.ceil(totalItems / pageSize);
-    const isFirstPage = activePage === 1;
-    const isLastPage = activePage === totalPages;
+    let totalPages = Math.ceil(totalItems / pageSizeLocal);
```

Comment:
to avoid confusion over `pageSize` and `showPageSizeChanger` being specified at the same time, let's establish that that the `showPageSizeChanger` prop takes precedence over `pageSize`

if `showPageSizeChanger` is true, use `selectedOption.value` as the divisor, else use `pageSize`